.github/workflows/check-branch.yml

```diff
@@ -0,0 +1,19 @@
+name: Enforce PR Branch Naming
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  check-branch-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fail if PR is from 'main' branch
+        run: |
+          if [[ "${{ github.head_ref }}" == "main" ]]; then
```

Comment:
I'm not a security expert, but I think there may be a risk of template injection here. 
`github.head_ref` is user-controllable input IIUC. I can't come up with something that would do harm right of the bat, but I can break the workflow by using a branch name like `foo";hostname`

Might be safer to have it be put in an env variable (outside of `run`) as to not influence script generation.